tests/rptest/tests/test_si_cache_space_leak.py

```diff
@@ -0,0 +1,143 @@
+
+    def __init__(self, test_context, *args, **kwargs):
+        test_name = test_context.test_name
+        si_params = self.test_defaults.get(
+            test_name) or self.test_defaults.get('default')
+        si_settings = SISettings(**si_params)
+        self._segment_size = si_params['log_segment_size']
+        extra_rp_conf = {
+            'disable_metrics': True,
+            'election_timeout_ms': 5000,
+            'raft_heartbeat_interval_ms': 500,
+            'segment_fallocation_step': 0x1000,
+            'cloud_storage_enable_remote_read': True,
+            'cloud_storage_enable_remote_write': True,
+            'retention_bytes': self._segment_size,
+        }
+        super().__init__(test_context,
+                         num_brokers=3,
+                         extra_rp_conf=extra_rp_conf,
+                         si_settings=si_settings)
+        self._ctx = test_context
+        self._verifier_node = test_context.cluster.alloc(
+            ClusterSpec.simple_linux(1))[0]
+        self.logger.info(
+            f"Verifier node name: {self._verifier_node.name}, segment_size: {self._segment_size}"
+        )
+
+    def init_producer(self, msg_size, num_messages):
+        self._producer = FranzGoVerifiableProducer(self._ctx, self.redpanda,
+                                                   self.topic, msg_size,
+                                                   num_messages,
+                                                   [self._verifier_node])
+
+    def init_consumer(self, msg_size, num_messages, concurrency):
+        self._consumer = FranzGoVerifiableRandomConsumer(
+            self._ctx, self.redpanda, self.topic, msg_size, num_messages,
+            concurrency, [self._verifier_node])
+
+    def free_nodes(self):
+        super().free_nodes()
+        wait_until(lambda: self.redpanda.sockets_clear(self._verifier_node),
+                   timeout_sec=120,
+                   backoff_sec=10)
+        self.test_context.cluster.free_single(self._verifier_node)
+
+    def validate_disk_space(self):
+        pass
+
+    @cluster(num_nodes=4)
+    @matrix(message_size=[10000],
+            num_messages=[100000],
+            num_read=[1000],
+            concurrency=[2])
+    def test_si_cache(self, message_size, num_messages, num_read, concurrency):
+
+        self.logger.info(f"Environment: {os.environ}")
+        if os.environ.get('BUILD_TYPE', None) == 'debug':
+            self.logger.info(
+                "Skipping test in debug mode (requires release build)")
+            return
+
+        self.init_producer(message_size, num_messages)
+        self._producer.start(clean=False)
+
+        def s3_has_some_data():
+            objects = list(self.redpanda.get_objects_from_si())
+            total_size = 0
+            for o in objects:
+                total_size += o.ContentLength
+            return total_size > self._segment_size
+
+        wait_until(s3_has_some_data, timeout_sec=300, backoff_sec=5)
```

Comment:
is this `wait_until` ensuring:

    The retention on the topic has to be
     small enough in order for SI to be involved.

Reply:
it ensures that the SI will be used